git_machete/tests/functional/test_machete.py

```diff
@@ -672,3 +680,682 @@ def test_help(self) -> None:
                         self.fail(f'Unexpected exception raised: {e}')
                     else:
                         self.fail('SystemExit expected but not raised')
+
+    def test_go_up(self) -> None:
+        """Verify behaviour of a 'git machete go up' command.
+
+        Verify that 'git machete go up' performs 'git checkout' to the
+        parent/upstream branch of a current branch.
+
+        """
+        (
+            self.repo_sandbox.new_branch("level-0-branch")
+            .commit()
+            .new_branch("level-1-branch")
+            .commit()
+        )
+        self.launch_command("discover", "-y")
+
+        self.launch_command("go", "up")
+
+        self.assertEqual(
+            'level-0-branch',
+            self.launch_command("show", "current").strip(),
+            msg="Verify that 'git machete go up' performs 'git checkout' to "
+                "the parent/upstream branch of a current branch."
+        )
+
+    def test_go_down(self) -> None:
+        """Verify behaviour of a 'git machete go down' command.
+
+        Verify that 'git machete go down' performs 'git checkout' to the
+        child/downstream branch of a current branch.
+
+        """
+        (
+            self.repo_sandbox.new_branch("level-0-branch")
+            .commit()
+            .new_branch("level-1-branch")
+            .commit()
+            .check_out("level-0-branch")
+        )
+        self.launch_command("discover", "-y")
+
+        self.launch_command("go", "down")
+
+        self.assertEqual(
+            'level-1-branch',
+            self.launch_command("show", "current").strip(),
+            msg="Verify that 'git machete go down' performs 'git checkout' to "
+                "the child/downstream branch of a current branch."
+        )
+
+    def test_go_first_root_with_downstream(self) -> None:
+        """Verify behaviour of a 'git machete go first' command.
+
+        Verify that 'git machete go first' performs 'git checkout' to
```

Comment:
Hmmm since the message in assertion seems to be identical to the function doc for all tests... maybe putting the message in doc is redundant then?

Reply:
No entirely. Doc describes whole test method, whereas assertion describe purpose of a particular oracle.

See below case with multiple assertions/oracles.
![image](https://user-images.githubusercontent.com/13608913/129091693-7bc9e0f3-8190-48e5-bb12-26b0bf9092c0.png)
![image](https://user-images.githubusercontent.com/13608913/129091554-3dbd487c-3ac5-4a87-ad3f-0fc642e5c7ca.png)

If you prefer that I can remove it from doc, but I think it could help someone get a gist of what is tested by just reading docstring and not looking into the code unnecessarly.

Reply:
Okay! right, makes sense &mdash; let's leave as is